### ISSUE TITLE
Stability issues

### ISSUE THREAD

**Issue by lapsio** [verse/verse-blender#12]
Hello. I'm trying to deploy some VM with Blender with Verse support for our dev team. We're working on game assets in small team (6 people). Verse sounds like heaven for working on landscapes but I'm experiencing terrible issues with stability.

1. Blender crashing
2. Server crashing
3. Blender randomly derps and one client  starts to freak out and not update anything while still receiving updates from others. Disconnect or unsubscribe throws python errors.
4. Sometimes blender enters weird state where it can neither Connect nor Disconnect

I'm not sure if it's problem with VM itself (I had hard time compiling Blender on Kubuntu, I don't typically use Debian based distros) because it's kind of possible that Blender in this VM is unstable itself but It's also possible that Verse itself is really unstable. We're working on low poly meshes so I don't think Verse should be overhelmed yet still sharing any more complex mesh (like 7k tris) usually results in various crashes.

May I ask you what environment do you use for development and if it's stable? Our team really hopes we could use Verse. After all we even considered switching back to blender 2.3/2.4 just to get Verse 1.0 support but it'd be much better if we could use something post 2.50. I can spend few days trying to make it work but I need to know if it makes sense because it's not possible to make this thing even remotely close to stable then I'm afraid it's problem we can't deal with.

Atm we experienced crashes every 2-5 minutes on average. We don't use single user on many instances, everyone has own user to avoid trouble.

**Comment by lapsio** [comment]
We're going to work over WAN. Info about bandwidth requirements would be also nice.

**Comment by lapsio** [comment]
https://dl.dropboxusercontent.com/u/44131220/verse.mkv

Here's some recording I wanted to present to team as demo but after some time I realized 30 minutes of crashes are bad promo. Our project is open source as well If you'd like to download assets used on recording they can be found here: https://github.com/PonyKart/PonyKart-UE4/tree/master/PonyKart%20Development%20Assets/Model%20assets

I don't know if it's problem on my side or with verse but I'd like to help somehow.

**Comment by jirihnidek** [comment]
Hi, thanks for your feedback and interest :-). I know it is not very stable, though. This Add-On is experiment and it reaches the edge what is possible to implement in Blender using Python. Keep in mind that this Add-On is not intended for production! When you wanted to work in collaborative environment, then try to use this: https://clara.io/

Git links works, but you have to use SSH (I added https link to readme.md too).

**Comment by lapsio** [comment]
Well clara.io has this one fundamental issue most of academic projects have - it's using brand new, custom, completely useless software with implemented this one feature (collab) right and somehow hoping it's enough to make it competitive with 3d software developed for over 15 years like Blender, 3ds, Maya, ZBrush... Most of people in our team have at least 8 years experience in 3d development so proof of concept software can't be used in production for us because of joke functionality. At this point it doesn't really matter if it's stable or not.

Anyways - do you maybe know if it's crashing that much because of Blender plugin implementation or verse library itself? Basically it looks like all crashes happen after sharing complex objects. As long as we're not sharing existing meshes but starting from cubes / planes it doesn't crash that often. Maybe it's not lost yet.

It doesn't "have to" work with super complex scenes after all. We though for a while how we can underload verse still preserving "presence" of second person on stage and quite reasonable idea is to work on NFS share with 3 files - A, B and base. People work on A and B, base is linked to both. After objects are done they're merged to base. This way we can have complex scenes while verse has to deal only with few objects like 2-3. Up to 2 per user.

What actually matters for us is not literal collab on single objects because there's no really point in that but being able to see what other users are adding to file now to avoid "collisions". Basically something like cross linking user groups but without constant file reloading required. 

Another approach we tried was to actually work in single Blender hosted on central server. But whereas Linux itself has pretty nice support for multiple mouse pointers/keyboards and there's xpra so technically should work - Blender doesn't seem to deal properly with multiple I/o at all... So didn't work well. We're much more likely to try to artificially force serious software to allow limited collab than try some poc soft.

**Comment by jirihnidek** [comment]
When I tested this Add-On, then I noticed some crashes, but it wasn't too often.

BTW: did you used Blender compiled with this patch: https://developer.blender.org/T42865. Keep in mind that saving blend files with such version can break reading blend files with official Blender release.

Other note: Verse library and Verse server prints to stdout lot of debug prints by default, when it is switched off or redirected to `/dev/null`, then it can be faster, when you works with big meshes.

I will repeat it once again. :-) This Add-On is not intended for production ATM. For this reason I do not propagate or announce it anywhere.

When Clara.io is not the right tool, then I would try Blender Cloud, Mercurial, etc.
